features/192-asset-gear/index.md

```diff
@@ -0,0 +1,667 @@
+# Asset Gear
+
+## Goals
+
+- Store user-generated asset to fulfill application needs.
+  - Photos and short videos taken by the application.
+  - Images and short videos chosen from the library.
+  - Artwork by the user.
+  - Attachments in a chat application.
+  - Documents of reasonable size.
+- Process image asset on the fly via query string.
+- Access control via presigned URL
+
+## Non-goals
+
+- Video live streaming such as Facebook Live.
+- File sharing application like Imgur.
```

Comment:
What does this meant? Why it is a non-goals compare with the goals? I don't get / see the difference.

Reply:
Such application typically has very special requirements on how the files are served. They may want to have their own rate limiting, analytics etc. I guess the Asset Gear is not targeting these applications.

Reply:
Sorry for not being clear. I was thinking of some requirements of such services may not be easily fulfilled by the Asset Gear. For example, some image hosting service checks the `Referer` header to ban hotlinking. Some file sharing service bans concurrent download. These features are not currently supported by the Asset Gear and I am not sure whether we should support them in the future. I wanted to emphasize that if the developer is building such applications, the current Asset Gear cannot offer what they need.

Reply:
@louischan-oursky noted... I guess my major issue with this is the non-goal's use cases is quite similar from a end-user perspective for "user static assets" and "file sharing application", eseentially user static assets == file sharing applications right? E.g. you used the sample of imgur, which is actually a perfect example of usage for user assets ("Photos and short videos taken by the application" == "imgur")

All examples of features you mentioned that we don't have right now... sounds like more coz we don't want to implement it due to time and maintenance reason, but not coz it is a different use case for different audience...

(a.k.a. Unlike Video Streaming or Static File Hosting which can be easily distinglished)

I would suggest just remove it unless we have a more clear way to distinglish the use cases...

Reply:
OK. Let me remove the second item in Non-goals.